File: src/main/java/com/ezypay/test/service/SubscriptionService.java
package com.ezypay.test.service;

import com.ezypay.test.dto.SubscriptionDto;
import java.util.List;

public interface SubscriptionService {
  SubscriptionDto getSubscriptionById(Integer userId);
  void saveSubscription(SubscriptionDto subscriptionDto);
  List< SubscriptionDto > getAllSubscriptions();
}

File: src/main/resources/static/js/app/SubscriptionService.js
'use strict'

angular.module('test.services', []).factory('SubscriptionService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getSubscriptionById = function(subscriptionId) {
				var url = CONSTANTS.getSubscriptionByIdUrl + subscriptionId;
				return $http.get(url);
			}
			service.getAllSubscriptions= function() {
				return $http.get(CONSTANTS.getAllSubscriptions);
			}
			service.saveSubscription = function(subscriptionDto) {
				return $http.post(CONSTANTS.saveSubscription, subscriptionDto);
			}
			return service;
		} ]);